src/xo-app/host/tab-network.js

```diff
@@ -118,56 +129,136 @@ class PifItem extends Component {
+        body: <ConfigureIpModal pif={this.props.pif} />
+      }).then(
+        params => {
+          if (!params.ip || !params.netmask) {
+            error(_('configIpErrorTitle'), _('configIpErrorMessage'))
+            return
+          }
+          return reconfigurePifIp(this.props.pif, { mode, ...params })
+        },
+        noop
+      )
+    }
+    return reconfigurePifIp(this.props.pif, { mode })
+  }
   render () {
-    const { networks, pif, vifsByNetwork } = this.props
+    const {pif} = this.props
     const { configModes } = this.state
-
+    return <Select
+      onChange={this._configIp}
+      options={configModes}
+      value={pif.mode}
+    >
+      {pif.mode}
+    </Select>
+  }
+}
+@connectStore(() => ({
+  vifsByNetwork: createGetObjectsOfType('VIF').groupBy('$network')
+}))
+class PifItemInUse extends Component {
+  render () {
+    const {networks, pif, vifsByNetwork} = this.props
     const pifInUse = some(vifsByNetwork[pif.$network], vif => vif.attached)
+    return <div className='text-xs-center'>
+      {_toggleDefaultLockingMode(
+        <Toggle
+          disabled={pifInUse}
+          onChange={() => editNetwork(pif.$network, { defaultIsLocked: !networks[pif.$network].defaultIsLocked })}
+          value={networks[pif.$network].defaultIsLocked}
+        />,
+        pifInUse && _('pifInUse')
+      )}
+    </div>
+  }
+}
 
-    return <tr>
-      <td>{pif.device}</td>
-      <td>{networks[pif.$network].name_label}</td>
-      <td>
-        {pif.vlan === -1
-          ? 'None'
-          : <Number value={pif.vlan} onChange={this._editPif}>
-            {pif.vlan}
-          </Number>
-        }
-      </td>
-      <td>
-        {pif.ip}
-        {' '}
-        {pif.ip && <a className='hidden-md-down' onClick={this._onEditIp} style={EDIT_BUTTON_STYLE}>
-          <Icon icon='edit' size='1' fixedWidth />
-        </a>}
-      </td>
-      <td>
-        <Select
-          onChange={this._configIp}
-          options={configModes}
-          value={pif.mode}
-        >
-          {pif.mode}
-        </Select>
-      </td>
-      <td><pre>{pif.mac}</pre></td>
-      <td>{pif.mtu}</td>
-      <td className='text-xs-center'>
-        {_toggleDefaultLockingMode(
-          <Toggle
-            disabled={pifInUse}
-            onChange={() => editNetwork(pif.$network, { defaultIsLocked: !networks[pif.$network].defaultIsLocked })}
-            value={networks[pif.$network].defaultIsLocked}
-          />,
-          pifInUse && _('pifInUse')
-        )}
-      </td>
-      <td>
+const COLUMNS = [
+  {
+    itemRenderer: (pif, networks) => {
+      return pif.device
+    },
+    default: true,
+    name: _('pifDeviceLabel'),
+    sortCriteria: 'device'
+  },
+  {
+    itemRenderer: (pif, networks) => {
+      return networks[pif.$network].name_label
+    },
+    name: _('pifNetworkLabel'),
+    sortCriteria: (pif, networks) => networks[pif.$network].name_label
+  },
+  {
+    itemRenderer: (pif, networks) => {
```

Comment:
You can use `component` instead of `itemRenderer`:

```js
component: PifItemVlan
```

The `pif` will be passed as `props.item` and `networks` as `props.userData`.

Reply:
Pierre told me that 'component' is used only when a big component is render, for that case, itemRender is enough.

Reply:
`PifItemVlan` *is* that "big component".

Reply:
Effectively, while it is connected to the store as you told me :+1: